### ISSUE TITLE
VARA on a Pi 3B+

### ISSUE THREAD

**Issue by km4ack** [WheezyE/Winelink#35]
Just an FYI. Below is an email I received from Jerry, K4OAM. He doesn't have a github account so I am just posting for him. This is not my work. 

Here is what I did to make VARA install and work on the Pi3B+ 32-bit Bullseye
version (3.2.2) of Build-A-Pi:

1. Temporarily increase the SWAP size to 750MiB
2. Download and install Pi-Apps from their website (tip from you previously)
3. From the Pi-Apps menu, request an install of Wine
4. At this point, a warning will appear that Wine cannot be installed on the
    existing kernel (2GB/2GB), with the option to build a new 3G/1B kernel.
5. Two 3G/1B options are presented:  Use an existing, but maybe outdated
    build (faster), or go out to the kernel source site and build a new, up-to-date
    kernel (verrrrry  sloooow)
6. I opted for a new, up-to-date build from the menu, and let it build overnight.
7. After a full 3G/1G build success message in the morning, I rebooted the Pi.
8. All Build-A-Pi programs seemed to work well, as before.
9. I then followed (step-by-step) your VARA install instrucable, from the top.
10.  All steps completed EXACTLY as you described, albeit slowly, based
       upon the fact this is a Pi3B+.
11.  At completion, VOILA, a working VARA HF and VARA FM on my lowly
      32-bit system.
12  Hope this info is useful.

**Comment by WheezyE** [comment]
Amazing, thank you for sending me this proof-of-concept report with the Pi3.  That's very heartening to hear that VARA HF/FM aren't too slow on that system to be of use.  And it's great leg work to have somebody done it the 'long way' for me already.

I've been eying BotSpot's kernel switcher code for a while but had never been at a stage where I felt the Pi4 was stable enough take everything one step farther to the Pi3 (I might have even asked him if I could borrow that section and I think he said yes, but I need to ask again since it's been so long).

I can have the script auto-detect if a user is running Pi3 or Pi4. Once I get some code cobbled together, we'll have to decide if we want to have the script do the 'long way' or 'short way' method.

**Comment by WheezyE** [comment]
Thank you again for relaying this info.  I'd known for a while that switching the kernel with Botspot's/Itai's code was going to be necessary for running Wine on Pi3, but I hadn't considered increasing the swap file size.  My Pi3B+ kept freezing up for long periods of time during install until I tried that.  I think (on preliminary tests) that the swap file might've fixed that.

I'm still getting lots of crashes with RMS Express, but I'll let you know if I can get things more stable on the Pi3B+.

**Comment by km4ack** [comment]
Thanks for the update!

**Comment by WheezyE** [comment]
Got this working as-of 558bc21f3cd8b4d3bd5c56c152d27c645121584a!

This took a long time to figure out and I ran away from the problem a few times tbh.  In the process of getting it working though, I ended up also implementing new methods for Pi-ID, OS-ID, arch-ID, Pi3 kernel switch, Pi swapfile increase, update i386-wine :armhf dependencies for box86:armhf on aarch64, and implement a CPU/OS sorting tree (with different wine install steps on each branch). The script has grown a lot more mature through implementing another Pi model.

I've got all the parts now to start soldering a DRA-like USB card to try to diagnose DRA PTT issues with VARA.  Fingers crossed I get all the joints soldered in the right places
